File: internal/dtos/response/customization.go
package response

type SoundscapeResponse struct {
	ID        string  `json:"id"`
	UserID    *string `json:"user_id,omitempty"`
	Name      string  `json:"name"`
	Category  string  `json:"category"`
	FilePath  string  `json:"file_path"`
	StreamURL string  `json:"stream_url"`
	Icon      string  `json:"icon"`
	Volume    float64 `json:"volume"`
	IsSystem  bool    `json:"is_system"`
	CreatedAt string  `json:"created_at"`
	UpdatedAt string  `json:"updated_at"`
}

type CustomFontResponse struct {
	ID         string  `json:"id"`
	UserID     *string `json:"user_id,omitempty"`
	Name       string  `json:"name"`
	FontFamily string  `json:"font_family"`
	SourceType string  `json:"source_type"`
	FileURL    string  `json:"file_url,omitempty"`
	FontURL    string  `json:"font_url,omitempty"`
	IsSystem   bool    `json:"is_system"`
	CreatedAt  string  `json:"created_at"`
	UpdatedAt  string  `json:"updated_at"`
}

type CustomThemeResponse struct {
	ID          string  `json:"id"`
	UserID      *string `json:"user_id,omitempty"`
	Name        string  `json:"name"`
	BgColor     string  `json:"bg_color"`
	TextColor   string  `json:"text_color"`
	AccentColor string  `json:"accent_color"`
	CustomCss   string  `json:"custom_css"`
	IsSystem    bool    `json:"is_system"`
	CreatedAt   string  `json:"created_at"`
	UpdatedAt   string  `json:"updated_at"`
}

File: web/src/types/customization.ts
export interface Soundscape {
  id: string;
  user_id?: string | null;
  name: string;
  category: string;
  file_path: string;
  stream_url: string;
  icon: string;
  volume: number;
  is_system: boolean;
  created_at: string;
  updated_at: string;
}

export interface CustomFont {
  id: string;
  user_id?: string | null;
  name: string;
  font_family: string;
  source_type: "file" | "url";
  file_url?: string;
  font_url?: string;
  is_system: boolean;
  created_at: string;
  updated_at: string;
}

export interface CustomTheme {
  id: string;
  user_id?: string | null;
  name: string;
  bg_color: string;
  text_color: string;
  accent_color: string;
  custom_css: string;
  is_system: boolean;
  created_at: string;
  updated_at: string;
}

export interface UploadSoundscapePayload {
  name: string;
  category?: string;
  icon?: string;
  volume?: number;
  audio_url?: string;
  is_system?: boolean;
  file?: File;
}

export interface UploadFontPayload {
  name: string;
  font_family: string;
  source_type: "file" | "url";
  font_url?: string;
  is_system?: boolean;
  file?: File;
}

export interface CreateCustomThemePayload {
  name: string;
  bg_color: string;
  text_color: string;
  accent_color: string;
  custom_css?: string;
  is_system?: boolean;
}

export interface UpdateCustomThemePayload {
  name: string;
  bg_color: string;
  text_color: string;
  accent_color: string;
  custom_css?: string;
}

export interface CustomizationListParams {
  cursor?: string;
  limit?: number;
}
